src/test/scala/edu/ie3/simona/service/primary/PrimaryServiceProxySqlIT.scala

```diff
@@ -0,0 +1,199 @@
+  TriggerWithIdMessage
+}
+import edu.ie3.simona.ontology.messages.services.ServiceMessage.PrimaryServiceRegistrationMessage
+import edu.ie3.simona.ontology.messages.services.ServiceMessage.RegistrationResponseMessage.{
+  RegistrationFailedMessage,
+  RegistrationSuccessfulMessage
+}
+import edu.ie3.simona.ontology.trigger.Trigger.{
+  ActivityStartTrigger,
+  InitializeServiceTrigger
+}
+import edu.ie3.simona.service.primary.PrimaryServiceProxy.InitPrimaryServiceProxyStateData
+import edu.ie3.simona.service.primary.PrimaryServiceWorker.SqlInitPrimaryServiceStateData
+import edu.ie3.simona.test.common.AgentSpec
+import edu.ie3.simona.test.helper.TestContainerHelper
+import edu.ie3.util.TimeUtil
+import org.scalatest.BeforeAndAfterAll
+
+import java.util.UUID
+
+class PrimaryServiceProxySqlIT
+    extends AgentSpec(
+      ActorSystem(
+        "PrimaryServiceWorkerSqlIT",
+        ConfigFactory
+          .parseString("""
+                     |akka.loglevel="OFF"
+          """.stripMargin)
+      )
+    )
+    with ForAllTestContainer
+    with BeforeAndAfterAll
+    with TestContainerHelper {
+
+  override val container: PostgreSQLContainer = PostgreSQLContainer(
+    "postgres:14.2"
+  )
+
+  private val simulationStart =
+    TimeUtil.withDefaults.toZonedDateTime("2020-01-01 00:00:00")
+
+  private val schemaName = "public"
+
+  override protected def beforeAll(): Unit = {
+    // Copy sql import scripts into docker
+    val sqlImportFile = getMountableFile("timeseries/")
+    container.copyFileToContainer(sqlImportFile, "/home/")
+
+    Iterable(
+      "time_series_p.sql",
+      "time_series_pqh.sql",
+      "time_series_mapping.sql"
+    ).foreach { file =>
+      val res = container.execInContainer("psql", "-Utest", "-f/home/" + file)
+      res.getStderr shouldBe empty
+    }
+  }
+
+  override protected def afterAll(): Unit = {
+    container.stop()
+    container.close()
+  }
+
+  // function definition because postgres parameters are only available after initialization
+  private def sqlParams: SqlParams = SqlParams(
+    jdbcUrl = container.jdbcUrl,
+    userName = container.username,
+    password = container.password,
+    schemaName = schemaName,
+    timePattern = "yyyy-MM-dd HH:mm:ss"
+  )
+
+  "A primary service proxy with SQL source" should {
+    val scheduler = TestProbe("Scheduler")
+
+    val proxyRef = TestActorRef(
+      PrimaryServiceProxy.props(
+        scheduler.ref,
+        simulationStart
+      )
+    )
+
+    "initialize when given proper SQL input configs" in {
+      val initData = InitPrimaryServiceProxyStateData(
+        SimonaConfig.Simona.Input.Primary(
+          None,
+          None,
+          None,
+          sqlParams = Some(sqlParams)
+        ),
+        simulationStart
+      )
+
+      val triggerIdInit1 = 1L
+
+      scheduler.send(
+        proxyRef,
+        TriggerWithIdMessage(
+          InitializeServiceTrigger(initData),
+          triggerIdInit1,
+          proxyRef
```

Comment:
Just something I was wondering about: Do you know why we include the receiving party in the trigger that gets directly sent to that individual? Is this for debugging reasons ?

Reply:
No, currently it is not used anywhere. We can probably remove it, but I want to touch scheduling at some point in the future anyway.